GenerateMacroDefinitionsFile.cmake

```diff
@@ -30,6 +30,10 @@ FILE(APPEND ${GENERATED_G3_DEFINITIONS} "// CMake induced definitions below. See
 
 FOREACH(definition ${G3_DEFINITIONS} )
    FILE(APPEND ${GENERATED_G3_DEFINITIONS} "#define ${definition}\n")
+   if(definition MATCHES ".* .*")
```

Comment:
@jmelion Just only saw the original issue. See my comment there. The important thing is that both library and anything linking to it must have the same macros defined. That is what the generated header is for.